src/basic/audit-util.c

```diff
@@ -80,7 +124,18 @@ bool use_audit(void) {
                         if (!cached_use)
                                 log_debug_errno(errno, "Won't talk to audit: %m");
                 } else {
-                        cached_use = true;
+                        /* If we try and use the audit fd but get ECONNREFUSED, it is because
+                         * we are not in the initial user namespace, and the kernel does not
+                         * have support for audit outside of the initial user namespace. */
+                        r = try_audit_request(fd);
+                        if (r < 0) {
+                                cached_use = r != -ECONNREFUSED;
```

Comment:
why are other errors also not causing the feature to be disabled? Is there a specific reason for that? If so, it should be mentioned in the comment

Reply:
The idea is that we know the -ECONNREFUSED error was given because we are making the request from outside the initial user namespace, and we do not want to be too strict in disabling audit (see https://elixir.bootlin.com/linux/latest/source/kernel/audit.c#L1036). I try to make the comment more clear.